crates/core/src/host.rs

```diff
@@ -50,6 +50,12 @@ pub struct HostData {
     /// Secrets given to this provider at runtime
     #[serde(default)]
     pub secrets: HashMap<String, SecretValue>,
+    /// The public key xkey of the host, used for decrypting secrets
+    #[serde(default)]
+    pub host_xkey_public_key: String,
+    /// The private key xkey of the provider, used for decrypting secrets
+    #[serde(default)]
+    pub provider_xkey_private_key: String,
```

Comment:
Let me see how the serialization works, just because if we use secrecy for that value and it changes the way it's serialized then that might affect the Go provider deserializing

Reply:
`secrecy` is not _super_ happy here with trying to get the secret to serialize & deserialize, and I wonder if just implementing `Zeroize` would be a better use of resources

Reply:
Ahh it looks like `SerializableSecret` has to be implemented:

> To prevent exfiltration of secret values via serde, by default Secret<T> does not receive a corresponding [Serialize](https://docs.rs/serde/1.0.126/x86_64-unknown-linux-gnu/serde/ser/trait.Serialize.html) impl. If you would like types of Secret<T> to be serializable with serde, you will need to impl the [SerializableSecret](https://docs.rs/secrecy/0.8.0/secrecy/trait.SerializableSecret.html) marker trait on T.

Reply:
Yeah I did impl that, which is just a marker trait at least, but it still gave me some trouble with deserializing 🤔 With the fact that I had to impl cloneablesecret, debugsecret, zeroize, etc etc I was planning to just push a commit that ensured that these structs zeroized on drop. Thoughts?

Reply:
Yeah that's probably fine... IMO don't waste *too* much time on this, I think, we can do this in a follow up (and whoever does that follow up can take all the time in the world to find a way that works well before we release), too much other good code in here to get caught up on this!